### PR TITLE
feat(android): support local file path

### PR DESCRIPTION
**Pull request by jhen0409**
~~iOS:~~
~~- Check uri prefix to use `filePath:` or `dotLottieFilePath:`~~

Android:
- Check uri prefix to use `FileInputStream` or `ZipInputStream`
- lottie-android v6.2 includes https://github.com/airbnb/lottie-android/pull/2411 so we can use ZipInputStream for load local dotlottie file

Fixes: https://github.com/lottie-react-native/lottie-react-native/issues/1142

### CODE REVIEW

**Comment by jhen0409** [comment]
Alright, I've found that iOS may not require these additional changes as it works by default. However, we only need to implement these changes for Android to support local file.

File: packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt [hunk 1]
```diff
@@ -119,13 +123,31 @@ class LottieAnimationViewPropertyManager(view: LottieAnimationView) {
         }
 
         animationURL?.let {
-            view.setAnimationFromUrl(it, it.hashCode().toString())
+            if (it.startsWith('/') || it.startsWith("file://")) {
+                try {
+                    view.setAnimation(FileInputStream(File(it)), Integer.toString(it.hashCode()))
```
Comment (matinzd) [inline packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt:128]: Is `Integer.toString(it.hashCode())` for caching purposes?

File: packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt [hunk 2]
```diff
@@ -119,13 +123,31 @@ class LottieAnimationViewPropertyManager(view: LottieAnimationView) {
         }
 
         animationURL?.let {
-            view.setAnimationFromUrl(it, it.hashCode().toString())
+            if (it.startsWith('/') || it.startsWith("file://")) {
```
Comment (matinzd) [inline packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt:126]: Can you give me an example for starting with `/` ? I feel like this checking can be unreliable.

Comment (jhen0409) [reply packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt:126]: I think `startsWith('/')` is should reliable for unix file system. We could use `File.exists()` instead, which should be cleaner.

Comment (matinzd) [reply packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt]: Yes! Much better :) Thanks.

File: packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt [hunk 3]
```diff
@@ -24,6 +24,10 @@ import com.facebook.react.views.text.TextAttributeProps.UNSET
 import com.facebook.react.util.RNLog
 import java.lang.ref.WeakReference
 import java.util.regex.Pattern
+import java.util.zip.ZipInputStream
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
```
Comment (matinzd) [inline packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.kt:30]: We can remove this I guess.

**Review by matinzd — APPROVED** [review]

**Comment by matinzd** [comment]
Thanks for your contribution!

**Pull request merged**